ts/packages/ts/src/utils.ts

```diff
@@ -1,24 +1,56 @@
 import _ from 'lodash';
 import Mustache from 'mustache';
+import Ajv, { SchemaObject } from 'ajv';
 
-export const ERR = (
-  message: string,
-  { location = [], suggestion = '' }: { location?: string[]; suggestion?: string } = {},
-) => {
-  return `${message}${!_.isEmpty(location) ? ` at ${location.join('.')}` : ''}${suggestion ? `, ${suggestion}` : ''}`;
-};
+export class ConfigError extends Error {
+  constructor(
+    public reason: string,
+    public hint: string = '', // code?: string,
+    public scope: [string, string][] = [],
```

Comment:
This is fine I think, but **_only_** because a golang version hasn't been tagged yet. 

Why? Because in the Golang version this ConfigError is already a `struct` (closes golang has to a class) and therefore things like `message`, `location` and `suggestion` are already publicly referenced by name. The cautionary tale here is to try to limit potential breaking changes (such as renaming things) as much as possible in OSS projects. Frequent breaking changes to key parts of the codebase are likely to put people off actually using the library, due to the pain of having to change their code every time they want to update dependencies. Ideally we should be aiming to make dependency version updates as seamless as possible.

I very much support making ConfigError a class here (and the equivalent in python), however I'd probably still keep reason/hint/scope named as message/suggestion/location. If you really do want to rename these, I'd suggest phasing it out over multiple versions, with a deprecation warning in between.

Reply:
@jamesL92, your caution regarding breaking changes and the importance of seamless updates is absolutely valid. I share the sentiment that we must be judicious with modifications that impact our users directly. Even though we're in a 0.X.Y version and the project is relatively young, which might afford us some flexibility with changes, I do acknowledge that each alteration requires thorough consideration. While this PR does suggest a significant shift for the ConfigSchema API, it’s a delicate balance to manage evolution while ensuring stability. Your suggestion of phased updates and deprecation warnings is a prudent approach that we should embed in our development process moving forward. I'll certainly bear this in mind with regard to the naming within ConfigError and our overall strategy for rolling out updates.

For now I suggest we take the "move fast and break things" for us to get to the first stable 1.X.Y version